landoapi/api/treestatus.py

```diff
@@ -0,0 +1,609 @@
+            )
+            status_change.trees.append(status_change_tree)
+
+        session.add(status_change)
+
+    session.commit()
+
+    return None, 204
+
+
+@result_object_wrap
+def get_tree(tree: str) -> dict:
+    """Handler for `GET /trees/{tree}`."""
+    result = get_tree_by_name(tree)
+    if result is None:
+        raise ProblemException(
+            404,
+            f"No tree {tree} found.",
+            "The tree does not exist.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404",
+        )
+    return combinedtree_as_dict(result)
+
+
+@auth.require_auth0(
+    groups=(auth.TREESTATUS_ADMIN), scopes=("lando", "profile", "email"), userinfo=True
+)
+def make_tree(tree: str, body: dict):
+    """Handler for `PUT /trees/{tree}`."""
+    session = db.session
+    if body["tree"] != tree:
+        raise ProblemException(
+            400,
+            f"Tree names must match, {tree} from url and {body['tree']} from body do not.",
+            "Tree names must match.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    new_tree = Tree(
+        tree=tree,
+        status=body["status"],
+        reason=body["reason"],
+        message_of_the_day=body["message_of_the_day"],
+    )
+    try:
+        session.add(new_tree)
+        session.commit()
+    except (sa.exc.IntegrityError, sa.exc.ProgrammingError):
+        raise ProblemException(
+            400,
+            f"Tree {tree} already exists.",
+            "Tree already exists.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    return new_tree.to_dict(), 200
+
+
+@auth.require_auth0(
+    groups=(auth.TREESTATUS_ADMIN), scopes=("lando", "profile", "email"), userinfo=True
+)
+def delete_tree(tree: str) -> tuple[str, int]:
+    """Handler for `DELETE /trees/{tree}`."""
+    remove_tree_by_name(tree)
+    return tree, 200
+
+
+@auth.require_auth0(
+    groups=(auth.TREESTATUS_USERS), scopes=("lando", "profile", "email"), userinfo=True
+)
+def update_log(id: int, body: dict):
+    """Handler for `PATCH /log/{id}`."""
+    session = db.session
+
+    tags = body.get("tags")
+    reason = body.get("reason")
+
+    if tags is None and reason is None:
+        return
+
+    # Update the log table.
+    update_tree_log(id, tags, reason)
+
+    # Iterate over all stack.
+    for status_change in StatusChange.query.all():
+        for tree in status_change.trees:
+            last_state = load_last_state(tree.last_state)
+
+            if last_state["current_log_id"] != id:
+                continue
+
+            if reason:
+                last_state["current_reason"] = reason
+            if tags:
+                last_state["current_tags"] = tags
+
+            tree.last_state = json.dumps(last_state)
+
+    session.commit()
+
+    return None, 200
+
+
+def get_logs_for_tree(tree_name: str, limit_logs: bool = True) -> list[dict]:
```

Comment:
Since this defaults to `True`, and uses a the default value (`TREE_SUMMARY_LOG_LIMIT`) might as well just change it to be an integer.
```suggestion
def get_logs_for_tree(tree_name: str, limit_logs: int = TREE_SUMMARY_LOG_LIMIT) -> list[dict]:
```